public/components/trace_analytics/components/traces/trace_table_helpers.tsx

```diff
@@ -21,7 +21,9 @@ import { TraceAnalyticsMode, TraceQueryMode } from '../../../../../common/types/
 import { appendModeToTraceViewUri, nanoToMilliSec } from '../common/helper_functions';
 
 export const fetchDynamicColumns = (columnItems: string[]) => {
-  return columnItems
+  const safeColumnItems = Array.isArray(columnItems) ? columnItems : [];
```

Comment:
why would the type be inaccurate and we need to check at runtime?

Reply:
Removed unnecessary type check as we control the input.